espresso/core/javatests/androidx/test/espresso/matcher/ViewMatchers1Test.java

```diff
@@ -571,6 +612,38 @@ public void onFocusChanged(
         is("view.getText() was \"text\" transformed text was \"text_transformed\""));
   }
 
+  @Test
+  public void withTextString_withTransformation_withSpecialCharacters_describeMismatch() {
+    TextView textView = new TextView(context);
+    textView.setText("text ");
+    textView.setTransformationMethod(
+        new TransformationMethod() {
+          @Override
+          public CharSequence getTransformation(CharSequence source, View view) {
+            return source + "transformed";
+          }
+
+          @Override
+          public void onFocusChanged(
+              View view,
+              CharSequence sourceText,
+              boolean focused,
+              int direction,
+              Rect previouslyFocusedRect) {
+          }
+        });
+    assertThat(
+        getMismatchDescription(withText(is("blah")), textView),
+        is(
+            "view.getText() was \"text[NBSP]\""
```

Comment:
Hmm... so this is going to replace any whitespace in the string... I'm not sure how many tests could possibly be broken by this change, if they are asserting on the error message in their tests/tools.

Reply:
Yes, indeed, but only if they contain special chars. 
That's what these changes aim for: to make such cases more human-readable.

Reply:
@adazh space, newline etc is not special, only "space-like creatures" are higlighted in IDEA and here it's only Non-breakable-space.

Also do you mean end-users would test like this?
```
try {
    onView(withId(R.id.foo)).check(matches(withText("abc"))
    fail()
} catch(ex) {
    assertThat(ex.message, equalTo("view.getText() was \"abc\""))
}
```

Reply:
> Also do you mean end-users would test like this?

Yep, I think it may be possible that they catch the exception and does a text check like this in their tests. And in this case, this change is a breaking change for them.